components/AgentForm.tsx

```diff
@@ -165,14 +173,21 @@ export default function BaseForm(props: Props) {
     iconUrl: props?.defaultValues?.iconUrl || defaultIconUrl,
   });
 
-  const upsertAgentMutation = useSWRMutation<
-    Prisma.PromiseReturnType<typeof upsertAgent>
-  >(`/api/agents`, postFetcher);
+  const getAgentQuery = useSWR<Prisma.PromiseReturnType<typeof getAgent>>(
+    `/api/agents/${router.query?.agentId}`,
+    fetcher
+  );
+  const agent = getAgentQuery?.data;
+  const agentConfig = agent?.interfaceConfig as AgentInterfaceConfig;
+  
+  const updateAgentMutation = useSWRMutation<
```

Comment:
This form can be used for creating / updating Agents
This implementation does not work for create.
Also, use variable names that work for both use cases
agentMutation instead of updateAgentMutation